src/gluonts/dataset/split/splitter.py

```diff
@@ -266,15 +259,73 @@ def _check_split_length(
     assert train_end + prediction_length <= test_end, msg
 
 
-class OffsetSplitter(pydantic.BaseModel, AbstractBaseSplitter):
+@dataclass
+class TestTemplate:
+    """
+    A class used for generating test data.
+
+    Parameters
+    ----------
+    splitter:
+        A specific splitter that knows how to slices training and
+        test data.
+    """
+
+    dataset: Dataset
+    splitter: AbstractBaseSplitter
+
+    def generate_instances(
+        self,
+        prediction_length: int,
+        windows=1,
+        distance=None,
+        max_history=None,
+    ):
+        """
+        Generate an iterator of test dataset, which includes input part and
+        label part.
+
+        Parameters
+        ----------
+        prediction_length
+            Length of the prediction interval in test data.
+        windows
+            Windows is used especially when doing ``rolling_split``, otherwise,
+            it should be ``1`` for a simple ``split``.
+        distance
+            Length of each window. If set to None, the prediction_length is
```

Comment:
This is rather the difference between the start of each test window generated, for each of the original dataset entries